### PR TITLE
Nft detail page + basic clio ws

### PR DESCRIPTION
**Pull request by shawnxie999**
### Context of Change
Add to .env
```
REACT_APP_CLIO_WS_PORT=51233
REACT_APP_CLIO_HOST=52.37.88.100
```

### Type of Change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Refactor (non-breaking change that only restructures code)
- [ ] Tests (You added tests for code that already exists, or your new feature included in this PR)
- [ ] Documentation Updates
- [ ] Release

### TypeScript/Hooks Update

- [ ] Updated files to React Hooks
- [ ] Updated files to TypeScript

## Before / After

image

### CODE REVIEW

**Comment by shawnxie999** [comment]
- some fields on detail page still filled with dummy values

- write test cases

File: public/locales/en-US/translations.json [hunk 1]
```diff
@@ -5,6 +5,7 @@
   "assets.no_issued_message": "No tokens found.",
   "assets.no_nfts_message": "No NFTs found.",
   "check_nft_id": "Please check your NFT ID",
+  "get_NFT_state_failed":"Unable to load NFT state",
```
Comment (ledhed2222) [inline public/locales/en-US/translations.json]: i think it should be `get_nft_state_failed`

File: src/containers/App/App.js [hunk 2]
```diff
@@ -46,6 +46,7 @@ const App = props => {
         `wss://${process.env.REACT_APP_P2P_RIPPLED_HOST}:${process.env.REACT_APP_RIPPLED_WS_PORT}`,
       ])
     : undefined;
+  socket.clioSocket = new XrplClient('ws://52.37.88.100:51233');
```
Comment (ledhed2222) [inline src/containers/App/App.js]: note - we'll need a more perm solution here. do not merge yet!

File: src/containers/NFT/NFTHeader/actions.js [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+import * as actionTypes from './actionTypes';
+
+export const loadNFTState = (tokenId, rippledSocket) => dispatch => {
+  if (!HASH_REGEX.test(tokenId)) {
+    dispatch({
+      type: actionTypes.NFT_STATE_LOAD_FAIL,
+      status: BAD_REQUEST,
+      error: '',
+    });
+    return Promise.resolve();
```
Comment (ledhed2222) [inline src/containers/NFT/NFTHeader/actions.js:12]: why wouldn't this be `Promise.reject`? Not sure what the pattern is for redux here.

File: src/containers/NFT/NFTHeader/index.js [hunk 4]
```diff
@@ -0,0 +1,246 @@
...
+import { loadNFTState } from './actions';
+import Loader from '../../shared/components/Loader';
+import '../../shared/css/nested-menu.css';
+import './styles.css';
+import SocketContext from '../../shared/SocketContext';
+import Tooltip from '../../shared/components/Tooltip';
+import Copy from '../../shared/components/Copy';
+
+class NFTHeader extends Component {
+  constructor(props) {
```
Comment (ledhed2222) [inline src/containers/NFT/NFTHeader/index.js]: we're transitioning away from class-based components and to functional components instead (IE hooks).

File: src/containers/NFT/NFTHeader/index.js [hunk 5]
```diff
@@ -0,0 +1,246 @@
...
+          {emailHash && (
+            <tr className="row">
+              <td className="col1">{t('email_hash')}</td>
+              <td className="col2">
+                {abbrvEmail}
+                <Copy className="copy" text={emailHash} />
+              </td>
+            </tr>
+          )}
+          {NFTTaxon !== undefined && (
```
Comment (ledhed2222) [inline src/containers/NFT/NFTHeader/index.js]: favor `!= null`

Comment (ledhed2222) [reply src/containers/NFT/NFTHeader/index.js]: or really just `{NFTTaxon && (...)}`

File: src/containers/NFT/NFTHeader/index.js [hunk 6]
```diff
@@ -0,0 +1,246 @@
...
+          {uri && (
+            <tr className="row">
+              <td className="col1">URI</td>
+              <td className="col2">
+                {abbrvURI}
+                <Copy className="copy" text={uri} />
+              </td>
+            </tr>
+          )}
+          {transferFee !== undefined && (
```
Comment (ledhed2222) [inline src/containers/NFT/NFTHeader/index.js]: why wouldn't there be a transfer fee?

File: src/containers/NFT/NFTHeader/index.js (hunk 5 above)
Comment (ledhed2222) [reply src/containers/NFT/NFTHeader/index.js]: actually why wouldn't there be a taxon?

File: src/containers/NFT/NFTHeader/index.js [hunk 7]
```diff
@@ -0,0 +1,246 @@
...
+        </tbody>
+      </table>
+    );
+  }
+
+  renderSettings() {
+    const { data } = this.props;
+    const { flags } = data;
+
+    const burnable = flags && flags.includes('lsfBurnable') ? 'enabled' : 'disabled';
```
Comment (ledhed2222) [inline src/containers/NFT/NFTHeader/index.js]: flags should always be present from the API

File: src/containers/NFT/NFTHeader/index.js [hunk 8]
```diff
@@ -0,0 +1,246 @@
...
+          </tr>
+        </tbody>
+      </table>
+    );
+  }
+
+  renderHeaderContent() {
+    const { t, data, language, tokenId } = this.props;
+    const { issuer } = data;
+    const abbrvIssuer = issuer && issuer.replace(/(.{23})..+/, '$1...');
```
Comment (ledhed2222) [inline src/containers/NFT/NFTHeader/index.js]: there should always be an issuer

File: src/containers/NFT/NFTHeader/actions.js (hunk 3 above)
Comment (shawnxie999) [reply src/containers/NFT/NFTHeader/actions.js:12]: the component picks up the state change of `status`, and renders error from there. dont think whichever way promise is resolved would matter

File: src/containers/NFT/NFTHeader/index.js (hunk 4 above)
Comment (shawnxie999) [reply src/containers/NFT/NFTHeader/index.js]: will change

File: src/containers/NFT/NFTHeader/index.js (hunk 7 above)
Comment (shawnxie999) [reply src/containers/NFT/NFTHeader/index.js]: im parsing the flag integer by returning an array of enabled flags. The ones that are disabled wont be inside the array, thus why the condition check

File: src/containers/NFT/NFTHeader/index.js (hunk 8 above)
Comment (ledhed2222) [reply src/containers/NFT/NFTHeader/index.js]: also i'm a bit confused as to what this regex is supposed to do. looks like you intended it to truncate characters 24 and beyond and replace them with `...`. This doesn't look right to me if that's the intention. Also, this is simple enough to do without regexes: `issuer.slice(0, 23).concat('...')`.

File: src/containers/NFT/NFTHeader/index.js [hunk 9]
```diff
@@ -0,0 +1,246 @@
...
+  render() {
+    const { loading, data } = this.props;
+    const { tooltip } = this.state;
+    const tokenId = data.NFTId;
+    return (
+      <div className="token-header">
+        <div className="section">
+          <div className="box-header">
+            <div className="token-title">
+              {!loading && <>NFT ID</>}
```
Comment (ledhed2222) [inline src/containers/NFT/NFTHeader/index.js]: why use a fragment here?

File: src/containers/NFT/index.js [hunk 10]
```diff
@@ -0,0 +1,114 @@
...
+  hints: ['check_nft_id'],
+};
+ERROR_MESSAGES.default = {
+  title: 'generic_error',
+  hints: ['not_your_fault'],
+};
+
+const getErrorMessage = error => ERROR_MESSAGES[error] || ERROR_MESSAGES.default;
+
+class NFT extends Component {
```
Comment (ledhed2222) [inline src/containers/NFT/index.js]: should use hooks not a class :)

File: src/containers/NFT/index.js [hunk 11]
```diff
@@ -0,0 +1,114 @@
...
+  hints: ['check_nft_id'],
+};
+ERROR_MESSAGES[BAD_REQUEST] = {
+  title: 'invalid_xrpl_address',
+  hints: ['check_nft_id'],
+};
+ERROR_MESSAGES.default = {
+  title: 'generic_error',
+  hints: ['not_your_fault'],
+};
```
Comment (ledhed2222) [inline src/containers/NFT/index.js:24]: why not build this as a literal in the first place?

```javascript
const ERROR_MESSAGES = {
  [NOT_FOUND]: {
    title: 'assest.no_nfts_message',
    hints: ['check_nft_id'],
  },
  ...,
};
```

File: src/containers/NFT/index.js [hunk 12]
```diff
@@ -0,0 +1,114 @@
...
+
+const getErrorMessage = error => ERROR_MESSAGES[error] || ERROR_MESSAGES.default;
```
Comment (ledhed2222) [inline src/containers/NFT/index.js]: you should use the new `??` operator here instead of `||`. The latter will treat `false` and `''` as false, but the former will only coalesce the specific values `null` and `undefined`

File: src/containers/NFT/styles.scss [hunk 13]
```diff
@@ -0,0 +1,54 @@
+@import '../shared/css/variables.scss';
+
+.token-page {
+  width: 100%;
+  .disclaimer-container {
+    display: flex;
+    flex-direction: column;
+    justify-content: space-between;
+    margin: 0px 32px;
+    margin-bottom: 0px;
```
Comment (ledhed2222) [inline src/containers/NFT/styles.scss]: this is unnecessary because of your `margin` rule above.

File: src/containers/NFT/styles.scss [hunk 14]
```diff
@@ -0,0 +1,54 @@
+@import '../shared/css/variables.scss';
+
+.token-page {
+  width: 100%;
+  .disclaimer-container {
+    display: flex;
+    flex-direction: column;
+    justify-content: space-between;
+    margin: 0px 32px;
```
Comment (ledhed2222) [inline src/containers/NFT/styles.scss]: people tend to prefer not to use units for a value of 0 in CSS: https://stackoverflow.com/questions/7923042/when-specifying-a-0-value-in-css-should-i-explicitly-mark-the-units-or-omit

File: src/containers/NFT/styles.scss [hunk 15]
```diff
@@ -0,0 +1,54 @@
...
+    margin-bottom: 0px;
+    @include for-size(tablet-landscape-up) {
+      width: 552px;
+      margin: 0px auto;
```
Comment (ledhed2222) [inline src/containers/NFT/styles.scss]: no `px`

File: src/containers/NFT/styles.scss [hunk 16]
```diff
@@ -0,0 +1,54 @@
...
+    }
+    @include for-size(desktop-up) {
+      flex-direction: row;
+    }
+
+    .disclaimer-left {
+      display: flex;
+      height: 15px;
+      align-items: end;
+      padding: 64px 0px 0px 5px;
```
Comment (ledhed2222) [inline src/containers/NFT/styles.scss]: no `px`

File: src/containers/NFT/styles.scss [hunk 17]
```diff
@@ -0,0 +1,54 @@
...
+      color: #9ba2b0;
```
Comment (ledhed2222) [inline src/containers/NFT/styles.scss]: is there a constant for this?

File: src/containers/NFT/styles.scss [hunk 18]
```diff
@@ -0,0 +1,54 @@
...
+      font-family: 'Work Sans', sans-serif;
```
Comment (ledhed2222) [inline src/containers/NFT/styles.scss:27]: is there a constant for this?

File: src/containers/NFT/styles.scss [hunk 19]
```diff
@@ -0,0 +1,54 @@
...
+      font-style: normal;
+      font-weight: 400;
+      line-height: 150%;
+    }
+
+    .disclaimer-right {
+      display: flex;
+      align-items: end;
+      color: #9ba2b0;
+      font-family: 'Work Sans', sans-serif;
```
Comment (ledhed2222) [inline src/containers/NFT/styles.scss:39]: constant?

File: src/containers/shared/components/Copy.tsx [hunk 20]
```diff
@@ -0,0 +1,33 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import copy from '../images/copy.png';
+
+interface Props {
+  text: string;
+  className: string;
```
Comment (ledhed2222) [inline src/containers/shared/components/Copy.tsx]: this is an optional property:

```typescript
interface Props {
  text: string
  className?: string
}
```

File: src/containers/shared/components/Copy.tsx [hunk 21]
```diff
@@ -0,0 +1,33 @@
...
+}
+
+const Copy = (props: Props) => {
+  const { text, className } = props;
```
Comment (ledhed2222) [inline src/containers/shared/components/Copy.tsx]: i think you can do this if you like (up to you)

```typescript
const Copy = ({ text, className }: Props) => {
  return (
    ...
  )
}
```

File: src/containers/shared/components/Copy.tsx [hunk 22]
```diff
@@ -0,0 +1,33 @@
...
+  );
+};
+Copy.propTypes = {
+  text: PropTypes.string,
+  className: PropTypes.string,
+};
+
+Copy.defaultProps = {
+  text: '',
+  className: '',
```
Comment (ledhed2222) [inline src/containers/shared/components/Copy.tsx]: should be `undefined` i believe

File: src/containers/shared/components/Copy.tsx [hunk 23]
```diff
@@ -0,0 +1,33 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import copy from '../images/copy.png';
+
+interface Props {
+  text: string;
+  className: string;
+}
+
+const Copy = (props: Props) => {
```
Comment (ledhed2222) [inline src/containers/shared/components/Copy.tsx]: i don't find the name of this component super descriptive. it could mean that it's to display "copy" as in "written content". what about something like "Copyable"?

File: src/containers/shared/css/tooltip.scss [hunk 24]
```diff
@@ -1,5 +1,17 @@
 @import './variables.scss';
 
+.nft {
+  position: absolute;
+  z-index: 1000;
```
Comment (ledhed2222) [inline src/containers/shared/css/tooltip.scss]: suspicious 🤔 . why did you need this?

File: src/containers/shared/css/tooltip.scss [hunk 25]
```diff
@@ -1,5 +1,17 @@
...
+  padding: 16px 16px 16px 16px;
```
Comment (ledhed2222) [inline src/containers/shared/css/tooltip.scss]: this should be written as `padding: 16px;`

File: src/containers/shared/css/tooltip.scss [hunk 26]
```diff
@@ -1,5 +1,17 @@
...
+  background: #a09f9f;
```
Comment (ledhed2222) [inline src/containers/shared/css/tooltip.scss]: constant?

File: src/containers/shared/css/tooltip.scss [hunk 27]
```diff
@@ -1,5 +1,17 @@
...
+  border-radius: 16px;
+  color: white;
```
Comment (ledhed2222) [inline src/containers/shared/css/tooltip.scss]: constant?

File: src/rippled/NFT.js [hunk 28]
```diff
@@ -0,0 +1,47 @@
...
+      flags,
+      transferFee,
+      issuer,
+      NFTTaxon,
+      NFTSequence,
+      uri,
+      validated,
+      status,
+      warnings,
+    };
```
Comment (ledhed2222) [inline src/rippled/NFT.js]: why not just `return formatNFTInfo(NFTInfo)`?

File: src/containers/NFT/NFTHeader/index.js (hunk 7 above)
Comment (ledhed2222) [reply src/containers/NFT/NFTHeader/index.js]: i get that, i'm saying why can't it be `const burnable = flags.includes('lsfBurnable') ? 'enabled': 'disabled'`

Comment (shawnxie999) [reply src/containers/NFT/NFTHeader/index.js]: the flags state is undefined until the ws request received the data.

File: src/containers/NFT/NFTHeader/index.js (hunk 8 above)
Comment (shawnxie999) [reply src/containers/NFT/NFTHeader/index.js]: but it does work on paper tho... i am actually just copying it from the existing token page 😂

File: src/containers/NFT/NFTHeader/index.js (hunk 9 above)
Comment (shawnxie999) [reply src/containers/NFT/NFTHeader/index.js]: fixed

File: src/containers/NFT/NFTHeader/index.js (hunk 8 above)
Comment (shawnxie999) [reply src/containers/NFT/NFTHeader/index.js]: > there should always be an issuer

the check here is to make sure its not undefined, since it will be undefined before the data arrives

File: src/containers/shared/components/Copy.tsx (hunk 21 above)
Comment (shawnxie999) [reply src/containers/shared/components/Copy.tsx]: done

File: src/containers/shared/css/tooltip.scss (hunk 24 above)
Comment (shawnxie999) [reply src/containers/shared/css/tooltip.scss]: yeah perhaps I shoudnt have done this. i added this because the new tooltip style is an entire rework of the current one, thus why it needs complete new styles. 

Should i adhere to the current tooltip style and just leave out the new styles for now? and then perhaps change the tooltips when the rework comes

File: src/containers/NFT/index.js (hunk 10 above)
Comment (shawnxie999) [reply src/containers/NFT/index.js]: done

File: src/containers/NFT/index.js (hunk 12 above)
Comment (shawnxie999) [reply src/containers/NFT/index.js]: thanks that makes sense! done

File: src/containers/NFT/index.js (hunk 11 above)
Comment (shawnxie999) [reply src/containers/NFT/index.js:24]: oops didnt know it. fixed!

File: src/containers/shared/components/Copy.tsx (hunk 20 above)
Comment (shawnxie999) [reply src/containers/shared/components/Copy.tsx]: fixed!

File: src/containers/NFT/NFTHeader/index.js [hunk 29]
```diff
@@ -0,0 +1,219 @@
...
+    setTooltip({ ...d, mode: 'nftId', x: event.pageX, y: event.pageY });
+  };
+
+  const hideTooltip = () => {
+    setTooltip(null);
+  };
+
+  const renderDetails = () => {
+    const { minted, domain, emailHash, NFTTaxon, uri, transferFee } = data;
+    const abbrvEmail = emailHash?.replace(/(.{20})..+/, '$1...');
```
Comment (ledhed2222) [inline src/containers/NFT/NFTHeader/index.js]: again i'm not sure this regex is right. also i think it's easier to read like so in this case:

```javascript
const abbrvEmail = emailHash?.slice(0, 20).concat('...')
```

File: src/containers/NFT/NFTHeader/index.js [hunk 30]
```diff
@@ -0,0 +1,219 @@
...
+    const abbrvURI = uri?.replace(/(.{20})..+/, '$1...');
```
Comment (ledhed2222) [inline src/containers/NFT/NFTHeader/index.js]: ditto

File: src/containers/NFT/NFTHeader/index.js [hunk 31]
```diff
@@ -0,0 +1,219 @@
...
+    return (
+      <table>
```
Comment (ledhed2222) [inline src/containers/NFT/NFTHeader/index.js:38]: not sure about using a table for this. @ckniffen what do you think?

File: src/containers/NFT/NFTHeader/index.js [hunk 32]
```diff
@@ -0,0 +1,219 @@
...
+            <div className="token-title">
+              NFT ID
+              <div className="token-type">
+                <div className="subscript">NFT</div>
+              </div>
+            </div>
+            <div
+              className="title-content"
+              onMouseOver={e => showTooltip(e, { tokenId })}
+              onFocus={() => {}}
```
Comment (ledhed2222) [inline src/containers/NFT/NFTHeader/index.js:163]: why do we need this noop?

File: src/containers/NFT/NFTHeader/styles.scss [hunk 33]
```diff
@@ -0,0 +1,401 @@
...
+    width: 1140px;
+  }
+}
+
+table {
+  margin-top: 24px;
+  font-size: 14px;
+
+  .col1 {
+    width: 180px;
```
Comment (ledhed2222) [inline src/containers/NFT/NFTHeader/styles.scss]: having explicit sizes like this is not responsive. We should be using flexbox to define a ratio

File: src/containers/NFT/NFTHeader/styles.scss [hunk 34]
```diff
@@ -0,0 +1,401 @@
...
+        font-size: 14px;
+        text-transform: uppercase;
+        @include semibold;
+      }
+      .value {
+        display: flex;
+
+        .value-content {
+          margin: auto;
+          color: #fff;
```
Comment (ledhed2222) [inline src/containers/NFT/NFTHeader/styles.scss]: constant?

File: src/containers/NFT/NFTHeader/styles.scss [hunk 35]
```diff
@@ -0,0 +1,401 @@
...
+          font-family: 'Work Sans', sans-serif;
```
Comment (ledhed2222) [inline src/containers/NFT/NFTHeader/styles.scss]: constant

File: src/containers/NFT/NFTHeader/styles.scss [hunk 36]
```diff
@@ -0,0 +1,401 @@
...
+      align-items: center;
+      justify-content: center;
+      order: 0;
+      padding: 2px 12px;
+      border: 1px solid #5f00e6;
+      margin-left: 8px;
+      background: #791aff;
+      border-radius: 30px;
+      box-shadow: 0px 1px 1px #5f00e5;
+      color: #fff;
```
Comment (ledhed2222) [inline src/containers/NFT/NFTHeader/styles.scss]: constant

File: src/containers/NFT/NFTHeader/styles.scss [hunk 37]
```diff
@@ -0,0 +1,401 @@
...
+      height: 22px;
+      box-sizing: border-box;
+      flex: none;
+      flex-direction: row;
+      flex-grow: 0;
+      align-items: center;
+      justify-content: center;
+      order: 0;
+      padding: 2px 12px;
+      border: 1px solid #5f00e6;
```
Comment (ledhed2222) [inline src/containers/NFT/NFTHeader/styles.scss]: constant

File: src/containers/NFT/NFTHeader/styles.scss [hunk 38]
```diff
@@ -0,0 +1,401 @@
...
+      margin-left: 8px;
+      background: #791aff;
```
Comment (ledhed2222) [inline src/containers/NFT/NFTHeader/styles.scss]: constant

File: src/containers/NFT/NFTHeader/styles.scss [hunk 39]
```diff
@@ -0,0 +1,401 @@
...
+      border-radius: 30px;
+      box-shadow: 0px 1px 1px #5f00e5;
```
Comment (ledhed2222) [inline src/containers/NFT/NFTHeader/styles.scss]: constant

File: src/containers/NFT/NFTHeader/styles.scss [hunk 40]
```diff
@@ -0,0 +1,401 @@
...
+      color: #fff;
+      font-family: 'Work Sans', sans-serif;
```
Comment (ledhed2222) [inline src/containers/NFT/NFTHeader/styles.scss]: constant

File: src/containers/NFT/NFTHeader/styles.scss [hunk 41]
```diff
@@ -0,0 +1,401 @@
...
+  .column {
+    display: inline-block;
+    width: 100%;
+    vertical-align: top;
+
+    @include for-size(desktop-up) {
+      width: 450px;
+
+      &.second {
+        width: calc(100% - 450px);
```
Comment (ledhed2222) [inline src/containers/NFT/NFTHeader/styles.scss]: you shouldn't need this with proper use of flexbox

File: src/containers/NFT/NFTHeader/styles.scss [hunk 42]
```diff
@@ -0,0 +1,401 @@
...
+    padding-bottom: 20px;
+  }
+
+  .column {
+    display: inline-block;
+    width: 100%;
+    vertical-align: top;
+
+    @include for-size(desktop-up) {
+      width: 450px;
```
Comment (ledhed2222) [inline src/containers/NFT/NFTHeader/styles.scss]: no width

File: src/containers/NFT/NFTHeader/styles.scss [hunk 43]
```diff
@@ -0,0 +1,401 @@
...
+            color: inherit;
+            text-decoration: none;
+            text-overflow: ellipsis;
+            text-transform: none;
+            vertical-align: middle;
+            white-space: nowrap;
+            @include medium;
+          }
+
+          b {
```
Comment (ledhed2222) [inline src/containers/NFT/NFTHeader/styles.scss]: i've never seen this element in use. prefer `<strong>` or a `<span>`

File: src/containers/NFT/NFTHeader/styles.scss [hunk 44]
```diff
@@ -0,0 +1,401 @@
...
+
+      &.xAddress ul {
+        width: 100%;
+
+        li {
+          display: flex;
+          justify-content: space-between;
+
+          @media (min-width: 450px) {
+            display: block;
```
Comment (ledhed2222) [inline src/containers/NFT/NFTHeader/styles.scss]: why?

File: src/containers/NFT/NFTHeader/styles.scss [hunk 45]
```diff
@@ -0,0 +1,401 @@
...
+          }
+
+          // .account {
+          //   overflow: hidden;
+          //   width: 285px;
+          //   padding-right: 10px;
+          //   text-overflow: ellipsis;
+          //   white-space: nowrap;
+          // }
```
Comment (ledhed2222) [inline src/containers/NFT/NFTHeader/styles.scss]: remember to remove

File: src/containers/NFT/NFTHeader/styles.scss [hunk 46]
```diff
@@ -0,0 +1,401 @@
...
+          }
+
+          // .weight {
+          //   text-align: right;
+          //   white-space: nowrap;
+          // }
+
+          // &.quorum {
+          //   display: block;
+          // }
```
Comment (ledhed2222) [inline src/containers/NFT/NFTHeader/styles.scss]: remember to remove

File: src/containers/NFT/NFTHeader/styles.scss [hunk 47]
```diff
@@ -0,0 +1,401 @@
...
+
+      &.signerList ul {
+        width: 100%;
+
+        li {
+          display: flex;
+          justify-content: space-between;
+
+          @media (min-width: 450px) {
+            display: block;
```
Comment (ledhed2222) [inline src/containers/NFT/NFTHeader/styles.scss]: why?

File: src/containers/shared/components/Copy.tsx [hunk 48]
```diff
@@ -0,0 +1,31 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import copy from '../images/copy.png';
+
+interface Props {
+  text: string;
+  className?: string;
+}
+
+const Copy = ({ text, className }: Props) => {
```
Comment (ledhed2222) [inline src/containers/shared/components/Copy.tsx]: Copyable?

File: src/containers/NFT/NFTHeader/styles.scss (hunk 40 above)
Comment (ckniffen) [reply src/containers/NFT/NFTHeader/styles.scss]: I dont even think this needs to be set.  Is there not a global setting of worksans?

File: src/containers/NFT/NFTHeader/index.js (hunk 31 above)
Comment (ckniffen) [reply src/containers/NFT/NFTHeader/index.js:38]: A `<dl>` is probably more apt but honestly everywhere is so inconsitent right now.

File: src/containers/NFT/NFTHeader/styles.scss (hunk 35 above)
Comment (ckniffen) [reply src/containers/NFT/NFTHeader/styles.scss]: Actually I think maybe the mockups have work sans being used but literarlly the rest of the site is Space Mono.  This should be done with other work.

File: src/containers/shared/components/Copy.tsx (hunk 48 above)
Comment (ckniffen) [reply src/containers/shared/components/Copy.tsx]: I like straight up CopyToClipboard.

File: src/containers/NFT/NFTHeader/styles.scss [hunk 49]
```diff
@@ -0,0 +1,230 @@
...
+        justify-content: center;
+        order: 0;
+        margin: auto;
+        text-align: center;
+        vertical-align: middle;
+        @include bold;
+      }
+    }
+
+    img {
```
Comment (ckniffen) [inline src/containers/NFT/NFTHeader/styles.scss:186]: is this for the copy to clipboard?
